android/library/src/main/java/com/liferay/mobile/screens/portlet/PortletDisplayScreenlet.java

```diff
@@ -0,0 +1,309 @@
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.mobile.screens.portlet;
+
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.util.AttributeSet;
+import android.view.LayoutInflater;
+import android.view.View;
+import com.liferay.mobile.android.auth.Authentication;
+import com.liferay.mobile.android.auth.basic.BasicAuthentication;
+import com.liferay.mobile.screens.R;
+import com.liferay.mobile.screens.base.BaseScreenlet;
+import com.liferay.mobile.screens.base.interactor.Interactor;
+import com.liferay.mobile.screens.context.LiferayScreensContext;
+import com.liferay.mobile.screens.context.LiferayServerContext;
+import com.liferay.mobile.screens.context.SessionContext;
+import com.liferay.mobile.screens.portlet.util.CssScript;
+import com.liferay.mobile.screens.portlet.util.InjectableScript;
+import com.liferay.mobile.screens.portlet.util.JsScript;
+import com.liferay.mobile.screens.portlet.view.PortletDisplayViewModel;
+import com.liferay.mobile.screens.util.AssetReader;
+import com.liferay.mobile.screens.util.LiferayLogger;
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URLEncoder;
+
+/**
+ * @author Sarai Díaz García
+ * @author Victor Galán Grande
+ */
+public class PortletDisplayScreenlet extends BaseScreenlet<PortletDisplayViewModel, Interactor>
+	implements PortletDisplayListener {
+
+	private boolean autoLoad;
+	private PortletDisplayListener listener;
+	private PortletConfiguration portletConfiguration;
+	private boolean isLoggingEnabled = true;
+
+	public PortletDisplayScreenlet(Context context) {
+		super(context);
+	}
+
+	public PortletDisplayScreenlet(Context context, AttributeSet attrs) {
+		super(context, attrs);
+	}
+
+	public PortletDisplayScreenlet(Context context, AttributeSet attrs, int defStyleAttr) {
+		super(context, attrs, defStyleAttr);
+	}
+
+	public PortletDisplayScreenlet(Context context, AttributeSet attrs, int defStyleAttr,
+		int defStyleRes) {
+		super(context, attrs, defStyleAttr, defStyleRes);
+	}
+
+	/**
+	 * Starts the operation and check if the Portlet URL it's correct and then loads it.
+	 * If the URL is not correct, the operation fails.
+	 */
+	public void load() {
+		getViewModel().showStartOperation(DEFAULT_ACTION);
+		if (portletConfiguration != null) {
+
+			String finalUrl = buildPortletUrl(portletConfiguration.getPortletUrl());
+			String body = buildBody();
+
+			String screensJs = new AssetReader(getContext()).read(R.raw.screens);
+
+			getViewModel().addScript(new JsScript("Screens.js", screensJs));
+
+			if (portletConfiguration.isThemeEnabled() && !portletConfiguration.getWebType()
+				.equals(PortletConfiguration.WebType.CUSTOM)) {
```

Comment:
invert the comparison: PortletConfiguration.WebType.CUSTOM.equals... (can't be null)